### ISSUE TITLE
integrate with surrealdb

### ISSUE THREAD

**Issue by duttaoindril** [serverpod/serverpod#294]
is serverpod database agnostic, or does it need to be set up for each database? can surrealdb be a backend that's supported?

**Comment by vlidholt** [comment]
It's possible to use Serverpod with any database, however, you will need Postgres for it to run. There are some packages for surrealdb on pub.dev. I haven't looked into them and do not know the quality of them though.

**Comment by fior-di-latte** [comment]
Can someone elaborate on why Postgres is needed in order to run any other database? Thanks in advance!:)

**Comment by yahu1031** [comment]
Yes, we need an example or a small documentation on how to connect with other SQL DBs.

**Comment by vlidholt** [comment]
> Can someone elaborate on why Postgres is needed in order to run any other database? Thanks in advance!:)

Some of Serverpod's features rely on the database. E.g., logging, future calls, health checks, default cloud storage, and authentication.

**Comment by marcelomendoncasoares** [comment]
Now that we have the `serverpod_database` package with the database interface properly abstracted and SQLite support, running other databases is only a matter of extending the supported database providers.
